plugins/resources/src/passthru.cpp

```diff
@@ -784,32 +789,36 @@ irods::error passthru_file_resolve_hierarchy(
                                  _curr_host,
                                  _out_parser,
                                  _out_vote );
- 
+
     double orig_vote = *_out_vote;
-    if ( irods::OPEN_OPERATION == ( *_opr ) ) {
-        double read_weight = 1.0;
-        irods::error ret = capture_weight(_ctx, READ_WEIGHT_KW, read_weight);
-        if ( !ret.ok() ) {
-            irods::log( PASS( ret ) );
-        }
-        else {
-            ( *_out_vote ) *= read_weight;
-            apply_weight_to_object_votes(_ctx, read_weight);
-        }
 
-    }
-    else if ( ( irods::CREATE_OPERATION == ( *_opr ) ||
-                irods::WRITE_OPERATION == ( *_opr ) ) ) {
-        double write_weight = 1.0;
-        irods::error ret = capture_weight(_ctx, WRITE_WEIGHT_KW, write_weight);
-        if ( !ret.ok() ) {
-            irods::log( PASS( ret ) );
+    try {
+        if (irods::OPEN_OPERATION == (*_opr)) {
+            double read_weight = 1.0;
+            if (const auto ret = capture_weight(_ctx, READ_WEIGHT_KW, read_weight); !ret.ok()) {
+                irods::log(PASS(ret));
+            }
+            else {
+                // NOLINTNEXTLINE(bugprone-narrowing-conversions,cppcoreguidelines-narrowing-conversions)
+                (*_out_vote) *= read_weight;
+                apply_weight_to_object_votes(_ctx, read_weight);
+            }
         }
-        else {
-            ( *_out_vote ) *= write_weight;
-            apply_weight_to_object_votes(_ctx, write_weight);
+        else if ((irods::CREATE_OPERATION == (*_opr) || irods::WRITE_OPERATION == (*_opr))) {
+            double write_weight = 1.0;
+            if (const auto ret = capture_weight(_ctx, WRITE_WEIGHT_KW, write_weight); !ret.ok()) {
+                irods::log(PASS(ret));
+            }
+            else {
+                // NOLINTNEXTLINE(bugprone-narrowing-conversions,cppcoreguidelines-narrowing-conversions)
+                (*_out_vote) *= write_weight;
+                apply_weight_to_object_votes(_ctx, write_weight);
+            }
         }
     }
+    catch (const irods::exception& e) {
+        log_resource::error("Error occurred while applying weight to vote: [{}]", e.client_display_what());
```

Comment:
I would print the name of the resource here, but getting the name of the resource may be the very thing which causes the exception!

Reply:
Perhaps you can capture it earlier in the function?

Reply:
Hmm... that's true. I'll play around with that.